flocker/node/_loop.py

```diff
@@ -332,11 +334,18 @@ def _send_state_to_control_service(self, state_changes):
             def record_acknowledged_state(ignored):
                 self._last_acknowledged_state = state_changes
 
-            d.addCallback(record_acknowledged_state)
+            def clear_acknowledged_state(failure):
+                # We don't know if the control service has processed the update
+                # or not. So we clear the last acknowledged state so that we
+                # always send the state on the next iteration.
+                self._last_acknowledged_state = None
+                return failure
```

Comment:
I think this is unrelated to the optimization.  It would have been nice to see this (and the related test changes) in a different PR so I could think about it independently.